Classes/Player/PlayerController.swift

```diff
@@ -76,22 +80,26 @@ class PlayerController: NSObject, Player, PlayerSettings {
     let reachability = PKReachability()
     var shouldRefresh: Bool = false
     
-    func prepare(_ config: MediaConfig) {
+    func setMedia(fromMediaEntry mediaEntry: MediaEntry) {
         // update the media source request adapter with new media uuid if using kaltura request adapter
-        self.updateRequestAdapterIfExists(forMediaConfig: config)
-        
-        self.currentPlayer.startPosition = config.startTime
-        self.assetBuilder = AssetBuilder(mediaEntry: config.mediaEntry)
-        self.assetBuilder?.build { (error: Error?, asset: AVAsset?) in
-            if let avAsset: AVAsset = asset {
-                self.currentPlayer.asset = avAsset
-                if DRMSupport.widevineClassicHandler != nil {
-                    self.addAssetRefreshObservers()
-                }
+        self.updateRequestAdapterIfExists(inMediaEntry: mediaEntry)
+        AssetBuilder.build(fromMediaEntry: mediaEntry) { error, asset in
+            if let assetToPrepare = asset {
+                self.assetToPrepare = assetToPrepare
+                self.onEventBlock?(PlayerEvent.SourceSelected(contentURL: assetToPrepare.url))
             }
         }
     }
     
+    func prepare(_ config: MediaConfig) {
```

Comment:
This shows a part of the problem. When you have "prepare" like that, you're lying to the user of the class: the MediaEntry inside this MediaConfig is ignored. 
Instead, you should save the MediaConfig as a member (inside setMedia), and use `self.mediaConfig.startPosition` here.